### PR TITLE
Bundle docker into the kpt image

### PR DESCRIPTION
**Pull request by ZhuGongpu**
Ref #226, bundle docker into the kpt image to enable `kpt fn run` in containers.

### CODE REVIEW

**Comment by ZhuGongpu** [comment]
@mortent @Liujingfang1 @monopole @frankfarzan

File: Dockerfile [hunk 1]
```diff
@@ -25,8 +25,10 @@ RUN go build -v -o /usr/local/bin/kpt ./
 
 FROM alpine:3.11
 RUN apk update && apk upgrade && \
-        apk add --no-cache git less man diffutils bash openssh && \
+    apk add --no-cache git less man diffutils bash openssh && \
     rm -rf /var/lib/apt/lists/* && \
     rm /var/cache/apk/*
+RUN apk --update add docker openrc && \
```
Comment (Liujingfang1) [inline Dockerfile:31]: What is the impact on image size when including docker?

Comment (ZhuGongpu) [reply Dockerfile:31]: Its size increases from `83.9MB` (`35.24MB` compressed) to `402MB` (`125.74MB` compressed).

Comment (wstrange) [reply Dockerfile:31]: Do you need all of docker, or just the client? Not sure if that is a seperate package or not..

Comment (mortent) [reply Dockerfile:31]: Maybe we should consider publishing a separate image that includes docker if we can't find a way to reduce the size?

Comment (ZhuGongpu) [reply Dockerfile:31]: @wstrange I found something [here](https://stackoverflow.com/a/43594065). It turns out the image size can be reduced into `150MB` (`52.81MB` compressed). @mortent 

#226 requires the docker client to be installed in the kpt container, and the docker client talks to the docker daemon through the mounted `docker.sock`. But I am not sure if there are any other cases that also need the docker daemon installed. @mortent Do you have any suggestions?

Comment (khrisrichardson) [reply Dockerfile:31]: @ZhuGongpu there's a separate docker-cli package, so you wouldn't necessarily need to download and unarchive the tgz

```bash
apk info docker-cli
docker-cli-19.03.11-r0 description:
Docker CLI

docker-cli-19.03.11-r0 webpage:
https://www.docker.io/

docker-cli-19.03.11-r0 installed size:
72921088
```

**Pull request merged**